tests/h/schemas/base_test.py

```diff
@@ -87,6 +116,90 @@ def test_it_returns_all_errors_in_message(self):
         assert message.startswith("'foo' is a required property, 'bar' is a required property")
 
 
+class TestRemoveUnknownProperties(object):
+
+    @pytest.mark.parametrize('data,expected', [
+        (
+            # data
+            {"foo": "string",
+             "baz": {
+                "foo": "string",
+                "bar": {
+                    "foo": "string"
+                }
+             }},
+            # expected
+            {"foo": "string",
+             "baz": {
+                "foo": "string",
+                "bar": {
+                    "foo": "string"
+                }
+             }},
+
+        ),
+        (
+            # data
+            {"foo": "string",
+             "no_exist": "string",
+             "baz": {
+                "foo": "string",
+                "no_exist": "string",
+                "bar": {
+                    "foo": "string",
+                    "no_exist": "string",
+                }
+             }},
+            # expected
+            {"foo": "string",
+             "baz": {
+                "foo": "string",
+                "bar": {
+                    "foo": "string",
+                }
+             }},
+
+        ),
+        (
+            # data
+            {"foo": "string",
+             "no_exist": {
+                "bar": "string",
+             }},
+            # expected
+            {"foo": "string"},
+
+        ),
+        (
+            # data
+            {"foo": "string",
+             "bar": {
+                "no_exist": "string",
+             }},
+            # expected
+            {"foo": "string"},
+
+        ),
+        (
+            # data
+            MultiDict([("foo", "string"),
+                       ("foo", "str"),
+                       ("baz", {"foo": "string"}),
```

Comment:
Is it actually possible to end up with a non-string value in `request.params`?

Reply:
No. But again-the intention here was to make it generic enough that other schemas could potentially use this function.